include/onnxruntime/core/platform/ort_mutex.h

```diff
@@ -3,11 +3,103 @@
 
 #pragma once
 #ifdef _WIN32
+#include <Windows.h>
 #include <mutex>
-#include <condition_variable>
 namespace onnxruntime {
-using OrtMutex = std::mutex;
-using OrtCondVar = std::condition_variable;
+// Q: Why OrtMutex is better than std::mutex
+// A: OrtMutex supports static initialization but std::mutex doesn't. Static initialization helps us prevent the "static
+// initialization order problem".
+
+// Q: Why std::mutex can't make it?
+// A: VC runtime has to support Windows XP at ABI level. But we don't have such requirement.
+
+// Q: Is OrtMutex faster than std::mutex?
+// A: Sure
+
+class OrtMutex {
+ private:
+  SRWLOCK data_ = SRWLOCK_INIT;
+
+ public:
+  constexpr OrtMutex() = default;
+  // SRW locks do not need to be explicitly destroyed.
+  ~OrtMutex() = default;
+  OrtMutex(const OrtMutex&) = delete;
+  OrtMutex& operator=(const OrtMutex&) = delete;
+  void lock() { AcquireSRWLockExclusive(native_handle()); }
+  bool try_lock() noexcept { return TryAcquireSRWLockExclusive(native_handle()) == TRUE; }
+  void unlock() noexcept { ReleaseSRWLockExclusive(native_handle()); }
+  using native_handle_type = SRWLOCK*;
+
```

Comment:
What's the benefit to having this 'using' statement?  I'm not a huge fan of type names that hide if something is a pointer, as it's harder to know whether the code using that type is correct (e.g. it's not immediately clear if native_handle() should return a reference instead of a copy).

I also don't quite understand why we need the native_handle() method. I wouldn't have expected anything outside of this class to need access to that, so the implementations of the various methods could just use &data_ instead couldn't they?

Reply:
All the function names and typedefs are copied from the C++ standard, so that our implementation can have the maximum compatibility with the standard. For example, you can use this class with std::lock_guard/std::unique_lock without worrying.